devicetype.go

```diff
@@ -0,0 +1,72 @@
+package balena
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"go.einride.tech/balena/odata"
+)
+
+const deviceTypeBasePath = "v6/device_type"
+
+// DeviceService handles communication with the device related methods of the
+// Balena Cloud API.
+type DeviceTypeService service
+
+type DeviceTypeResponse struct {
+	ID        uint64 `json:"id"`
+	Name      string `json:"name"`
+	Slug      string `json:"slug"`
+	IsPrivate bool   `json:"is_private"`
+	// Logo will only be populated when explicitly selected through a query parameter: `$select=logo`.
+	Logo                string       `json:"logo"`
+	IsOfCPUArchitecture odata.Object `json:"is_of__cpu_architecture"`
+}
+
+// Get returns information on a single device type given its ID or UUID.
```

Comment:
Does a device type have an UUID? If not maybe update the comment below?

```suggestion
// Get returns information on a single device type given its ID.
```

And if so, maybe change the ID type?